src/ts/components/sign-up/modal/index.tsx

```diff
@@ -0,0 +1,187 @@
+import React from 'react';
+import { Box, PseudoBox, Flex, Text } from '@blockstack/ui';
+// import { useHover } from 'use-events';
+import { Logo } from '@components/logo';
+import ChevronRightIcon from 'mdi-react/ChevronRightIcon';
+import CloseIcon from 'mdi-react/CloseIcon';
+import { Image } from '@components/image';
+import { BoxProps } from '@blockstack/ui/dist/box';
+
+interface ModalContextTypes {
+  isOpen: boolean;
+  doOpenModal?: () => void;
+  doCloseModal?: () => void;
+}
+
+const ModalContext = React.createContext<ModalContextTypes>({
+  isOpen: false,
+});
+
+const useModalState = () => React.useContext(ModalContext);
+
+interface HeaderTitleProps {
+  title: string;
+  hideIcon?: boolean;
+}
+
+const HeaderTitle: React.FC<HeaderTitleProps> = ({
+  hideIcon = false,
+  title,
+}) => (
+  <Flex align="center">
+    {hideIcon ? null : <Logo mr={2} />}
+    <Text fontWeight="bold" fontSize={'12px'}>
+      {title}
+    </Text>
+  </Flex>
+);
+
+const HeaderCloseButton: React.FC<BoxProps> = ({ onClick }) => (
+  <PseudoBox
+    color="ink.300"
+    opacity={0.5}
+    _hover={{ opacity: 1, cursor: 'pointer' }}
+    onClick={onClick}
+  >
+    <CloseIcon size={20} />
+  </PseudoBox>
+);
+
+interface AppIconProps {
+  src: string;
+  name: string;
+}
+
+const AppIcon: React.FC<AppIconProps> = ({ src, name, ...rest }) => (
+  <Box size={6} {...rest}>
+    <Image src={src} alt={name} title={name} />
+  </Box>
+);
+
+interface IModalHeader {
```

Comment:
Notice we're inconsistently using an `I` prefix. 
I'd vote against, they're treated as separate entities in TS so it's usually implicit what is/isn't a type contextually.

Reply:
Agreed - this was the original interface, but I also prefer to move away from `I` prefixes. That's present elsewhere in some places in the app, too.